src/carousel.tsx

```diff
@@ -334,22 +334,38 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
 
     const currentPageIndex = calcCurrentPageIndex(scrollLeft, pagesScrollPositions);
 
+    const pageChangeControl = React.useRef<{canEmit: boolean; lastPageIndex: number}>({
+        canEmit: false,
+        lastPageIndex: 0,
+    });
+
     React.useEffect(() => {
         if (onPageChange) {
-            const lastShownItemIndex = Math.min(
-                (currentPageIndex + 1) * itemsPerPageFloor - 1,
-                items.length - 1
-            );
-            const shownItemIndexes = [];
-            for (let i = 0; i < itemsPerPageFloor; i++) {
-                const idx = lastShownItemIndex - i;
-                if (idx >= 0) {
-                    shownItemIndexes.unshift(idx);
+            if (
+                pageChangeControl.current.canEmit &&
+                pageChangeControl.current.lastPageIndex !== currentPageIndex
+            ) {
+                const lastShownItemIndex = Math.min(
+                    (currentPageIndex + 1) * itemsPerPageFloor - 1,
+                    items.length - 1
+                );
+                const shownItemIndexes = [];
+                for (let i = 0; i < itemsPerPageFloor; i++) {
+                    const idx = lastShownItemIndex - i;
+                    if (idx >= 0) {
+                        shownItemIndexes.unshift(idx);
+                    }
                 }
+
+                onPageChange({pageIndex: currentPageIndex, shownItemIndexes});
             }
-            onPageChange({pageIndex: currentPageIndex, shownItemIndexes});
+
+            pageChangeControl.current = {
+                canEmit: !initialActiveItem || pageChangeControl.current.lastPageIndex !== currentPageIndex,
```

Comment:
This seems weird:
canEmit seems to have similar logic to line 346, only adding a condition of !initialActiveItem.

Isn't this redundant?

Reply:
Yes, I don't get this logic. Also, why `!initialActiveItem` ?

Reply:
when initialActiveItem is used, the carousel performs a scroll move to the initial item, this is not caused by the user swiping the carousel, so we don't want to fire a page changed in this case, so thats why I've used the !initialActiveItem.

Reply:
Its not redundant, as we need to avoid calling the onPageChange for this first automatic transition when initialActiveItem is set and different than 0.

Reply:
Let's say: initialActiveItem is A
Let's say: pageChangeControl.current.lastPageIndex !== currentPageIndex is B

canEmit is only used in line 345
canEmit = !A || B

and the condition is:
canEmit && B

so:
(!A || B) && B

so, that's the same as saying B by doing pageChangeControl.current = {...}

would it not make more sense to be explicit with this logic instead of ofuscating it with canEmit?

let's keep a ref like:
noPaginationHasOcurred = true

let's modify goToPage to treat the inital positioning differently
```
const goToPage = (page, animate, initialPositioning) =>  {
....
if (!initialPositioning){
  noPaginationHasOcurred = false;
}
...
}
```

Now the condition for when changing page can be more explicit
```
if (noPaginationHasOcurred) {
   return; // We don't emit events prior to first pagination
}

if (pageChangeControl.current.lastPageIndex !== currentPageIndex){
  ...we do stuff here...
}

```

Reply:
This doesn't works in the case the initial item index is 0 or undefined, be cause the goToPage function is called only when initial item index is set, different than 0, or the user clicks on an arrow.. (it doesnt get called on user swipe).